packages/runtime-dom/src/apiCustomElement.ts

```diff
@@ -581,18 +581,29 @@ export class VueElement
     owner?: ConcreteComponent,
   ) {
     if (!styles) return
+    const styleList: HTMLStyleElement[] = []
     if (owner) {
-      if (owner === this._def || this._styleChildren.has(owner)) {
+      if (owner === this._def) {
         return
       }
-      this._styleChildren.add(owner)
+      if (this._styleChildren.has(owner)) {
+        const styleList = this._styleChildren.get(owner)!
+        styleList.forEach(s => s.remove())
```

Comment:
This line is unnecessary. `prepend` will move nodes that already exist in the DOM tree.